src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -560,6 +560,22 @@ public override sbyte SByteValue
                 }
             }
 
+            public override short Int16Value
```

Comment:
>public override short Int16Value [](start = 12, length = 32)

Should we also override UInt16Value, Uint32Value and UInt64Value properties? For example, what value do we want to get from UInt16Value property for the following scenario (two different values are  printed)?
```
                byte x = byte.MaxValue; // ConstantValueTypeDiscriminator.SByte
                System.Console.WriteLine((ushort)x);
                System.Console.WriteLine((ushort)(sbyte)x);
```